### PR TITLE
fix: uri must not be null on mmscans, switched to ajax loading, fix 4…

### PR DESCRIPTION
**Pull request by alekc**
…04 on the last page

Fixes https://github.com/tachiyomiorg/tachiyomi-extensions/issues/16561

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: multisrc/overrides/madara/mmscans/src/MMScans.kt [hunk 1]
```diff
@@ -12,8 +18,76 @@ class MMScans : Madara("MMScans", "https://mm-scans.org", "en") {
...
+            latestUpdatesFromElement(element)
+        }
+
+        val hasNextPage = latestUpdatesNextPageSelector()?.let { selector ->
+            document.select(selector).first()
+        } != null
+
+        return MangasPage(mangas, !hasNextPage)
+    }
+
```
[the remark this answers is not in the document]
Comment (alessandrojean) [reply multisrc/overrides/madara/mmscans/src/MMScans.kt:51]: You can use something like `div.container:not(:has(.the-element))`

[the remark this answers is not in the document]
Comment (alekc) [reply multisrc/overrides/madara/mmscans/src/MMScans.kt:51]: ok, let me try that.

File: multisrc/overrides/madara/mmscans/src/MMScans.kt [hunk 2]
```diff
@@ -12,8 +18,76 @@ class MMScans : Madara("MMScans", "https://mm-scans.org", "en") {
...
+    fun madara_load_more(page: Int, meta_key: String): Request {
```
Comment (alessandrojean) [inline multisrc/overrides/madara/mmscans/src/MMScans.kt:52]: Please use camelCase.

```suggestion
    fun oldLoadMoreRequest(page: Int, metaKey: String): Request {
```

I also suggested a new name for the function as Madara used to do this call before, but was removed for security reasons.

File: multisrc/overrides/madara/mmscans/src/MMScans.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (alekc) [reply multisrc/overrides/madara/mmscans/src/MMScans.kt:51]: So, this is an example of the returned value:
```html
<div class="page-item-detail">
  <div id="manga-item-3343" class="item-thumb" data-post-id="3343">
    <a href="https://mm-scans.org/manga/chemy1243/" title="Chemy">
      <img width="350" height="476"  data-src="https://mm-scans.org/wp-content/uploads/2022/06/resource-2-350x476.jpg" data-srcset="https://mm-scans.org/wp-content/uploads/2022/06/resource-2-350x476.jpg 350w, https://mm-scans.org/wp-content/uploads/2022/06/resource-2-110x150.jpg 110w" data-sizes="(max-width: 350px) 100vw, 350px" class="img-responsive lazyload effect-fade" src="https://mm-scans.org/wp-content/themes/kronkyandnoahstheme/images/dflazy.jpg"  style="padding-top:476px;"  alt="resource (2)"/>
      <div class="list-shadow"></div>
      <div class="flag korea"></div>
      <div class="statut drop">Drop</div>
    </a>
  </div>
  <div class="item-summary">
    <a href="https://mm-scans.org/manga/chemy1243/">
      <h3>Chemy</h3>
    </a>
  </div>
</div>
<div class="page-item-detail">
  <div id="manga-item-1875" class="item-thumb" data-post-id="1875">
    <a href="https://mm-scans.org/manga/reversevillain1234/" title="Reverse Villain">
      <img width="350" height="469"  data-src="https://mm-scans.org/wp-content/uploads/2020/12/44575.png" data-srcset="https://mm-scans.org/wp-content/uploads/2020/12/44575.png 560w, https://mm-scans.org/wp-content/uploads/2020/12/44575-224x300.png 224w" data-sizes="(max-width: 350px) 100vw, 350px" class="img-responsive lazyload effect-fade" src="https://mm-scans.org/wp-content/themes/kronkyandnoahstheme/images/dflazy.jpg"  style="padding-top:469px;"  alt="44575"/>
      <div class="list-shadow"></div>
      <div class="flag unset"></div>
      <div class="statut complete">Completed</div>
    </a>
  </div>
  <div class="item-summary">
    <a href="https://mm-scans.org/manga/reversevillain1234/">
      <h3>Reverse Villain</h3>
    </a>
  </div>
</div>
<div class="invi no-posts">
  <!-- -->
</div>
```

as you can see they are all on the same level, so the :not :has would not work due to a lack of the parent container. Should we just leave the "original" (not the prettiest) solution for now?

[the remark this answers is not in the document]
Comment (alessandrojean) [reply multisrc/overrides/madara/mmscans/src/MMScans.kt:51]: Doesn't it work to use `:not(:has(.no-posts))` directly? IIRC Jsoup always wrap it into a body element when there's none so it should work.

File: multisrc/overrides/madara/mmscans/src/MMScans.kt (hunk 2 above)
Comment (alessandrojean) [reply multisrc/overrides/madara/mmscans/src/MMScans.kt]: Also need to rename the usages of it.

**Pull request merged**